tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -1352,6 +1303,17 @@ add_segmentby_to_equivalence_class(EquivalenceClass *cur_ec, CompressionInfo *in
 		if (context->current_col_info == NULL)
 			continue;
 
+		if (!COMPRESSIONCOL_IS_SEGMENT_BY(context->current_col_info))
```

Comment:
This was a bug, we didn't check if the column is a segmentby, and generated nonsensical EquivalenceMembers for compressed  columns.